### ISSUE TITLE
[rfe] add package to wrapper env without exposing tools

### ISSUE THREAD

**Issue by nanocell** [nerdvegas/rez#19]
I want to be able to do the following:

rez-env '(foo ~~bar)'

where the package 'bar' will be added to the wrapper shell but some special syntax (~~ in this case) will prevent its tools to be exposed to the main / master shell.

**Comment by bpabel** [comment]
I'm confused what you mean by this.  Both the `foo` and `bar` packages will not be available in the main/master shell.  They will both be available only in the subshell that `rez-env` creates.

**Comment by bpabel** [comment]
This issue should probably be closed.

**Comment by nanocell** [comment]
6 years later ... I can't even remember what this was about. Closing :).

**Comment by nerdvegas** [comment]
6 years?? I feel old.

So just to keep this on the radar. I think it'd be great to have this kind
of functionality - ie, allow packages to alter their behaviour in a
resolved env, based on user input (in Van's example above, this is a custom
syntax that means "don't expose this package's cli tools").

I've had in mind for some time working on a generalised version of this.
Basically, a resolve would be able to pass an arbitrary dict of so-called
"resolve params". Packages would have access to these params - both in
their commands() funcs and in late-bound attribs - and would be able to
change their definition accordingly. It's a blunt instrument, but I can
think of lots of cases where it'd be useful.

Not sure what the syntax would be just yet, but let's assume that we can
set params both globally, and specific to a package. You might then do
something like this:

    ]$ rez-env foo bah:cli=false

In other words the bah package would see the dict {"cli": False} and would
then have its commands section not expose any of its tools.The same
mechanism might be how we implement explicit variant selection, which has
been on the cards for a long time (there are cases where, in the presence
of non-mutually-exclusive variants, it is hard to control which variant is
resolved).

Cheers
A
